src/charm.py

```diff
@@ -1,103 +1,142 @@
 #!/usr/bin/env python3
-# Copyright 2024 Ali
+# Copyright 2024 Canonical Ltd.
 # See LICENSE file for licensing details.
-#
-# Learn more at: https://juju.is/docs/sdk
 
-"""Charm the service.
-
-Refer to the following tutorial that will help you
-develop a new k8s charm using the Operator Framework:
-
-https://juju.is/docs/sdk/create-a-minimal-kubernetes-charm
-"""
+"""Charm definition and helpers."""
 
 import logging
 
-import ops
+from log import log_event_handler
+from ops import main, pebble
+from ops.charm import CharmBase
+from ops.model import ActiveStatus, MaintenanceStatus
+from ops.pebble import CheckStatus
+from state import State
 
-# Log messages can be retrieved using juju debug-log
 logger = logging.getLogger(__name__)
 
-VALID_LOG_LEVELS = ["info", "debug", "warning", "error", "critical"]
+WEB_UI_PORT = 8080
 
 
-class AirbyteUiK8SOperatorCharm(ops.CharmBase):
-    """Charm the service."""
+class AirbyteK8SOperatorCharm(CharmBase):
+    """Charm the application."""
 
     def __init__(self, *args):
         super().__init__(*args)
-        self.framework.observe(self.on['httpbin'].pebble_ready, self._on_httpbin_pebble_ready)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self._state = State(self.app, lambda: self.model.get_relation("peer"))
+
+        self.name = "airbyte-webapp"
```

Comment:
perhaps: `self.app.name`

Reply:
`self.app.name` is the name that the user has deployed the application with, which could vary. Fixing the name of the container to `airbye-webapp` would be more suitable to have uniform instructions on accessing the container.